### DISCUSSION TITLE
Would adding an overload for a step size to `Enumerable.Range` be too niche of a request?

### DISCUSSION THREAD

**Discussion by tacosontitan** [dotnet/runtime#97790] [Polls]
I'm working on a project for code golf and I frequently use `Enumerable.Range` to generate data, but it's always lacked the ability to specify a custom step size. This leads to me using boilerplate like:

```cs
// proper-ish C# for this poll
var samples = new List<int>(capacity: 100);
for (int currentValue = 0, step = 5, count = 1; count < 100; count++) {
    samples.Add(currentValue);
    currentValue += step;
}
```

It would be nice to simplify this to:
```cs
var samples = Enumerable.Range(start: 0, count: 100, stepSize: 5);
```

However, this seems like a pretty niche overload, which doesn't make it a solid proposal IMHO.

**Comment by KennethHoff** [comment]
Not gonna lie, I've never thought about this being valid syntax.

```c#
for (int currentValue = 0, step = 5, count = 1; 
```

Anyways.. given that this is all you need, and the use-case is so highly niche, I don't see them adding it.

That said though, you can create your own struct with its own enumerator, so you can do something like this:

```c#
foreach (var i in new MySteppyRange(0, 100, 5))
{}
```

Which is not only less code (especially with a less dumb name), but it's also fully legal code already!

**Reply by tacosontitan** [reply]
I'm not opposed to just creating a static method for golfing support like `R(1,100,5)` which would be nice and short, but it got me thinking about the lack of official support for step size, so I decided to bring it up.

**Comment by stephentoub** [comment]
If we did add it, I expect it would be as part of adding a generic range method:
https://github.com/dotnet/runtime/issues/97689

**Reply by tacosontitan** [reply]
Now that's more like it 😄 thanks for the link!
